05/app.js

```diff
@@ -1,37 +1,62 @@
-const apiUrl = 'http://localhost:3000/users';
+const apiUrl = "http://localhost:3000/users";
 
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener("DOMContentLoaded", init);
 
 function init() {
-    loadUsers();
+  loadUsers();
+  addUser();
 }
 
 function loadUsers() {
-    const promise = fetchGet(apiUrl);
+  const promise = fetchGet(apiUrl);
 
-    promise
-        .then(data => insertUsers(data))
-        .catch(err => console.error(err));
+  promise.then((data) => insertUsers(data)).catch((err) => console.error(err));
 }
 
 function fetchGet(url) {
-    return fetch(url)
-        .then(resp => {
-            if(resp.ok) {
-                return resp.json();
-            }
-
-            return Promise.reject(resp);
-        });
+  return fetch(url).then((resp) => {
+    if (resp.ok) {
+      return resp.json();
+    }
+
+    return Promise.reject(resp);
+  });
+}
+
+function addUser() {
+  const formEl = document.querySelector("form");
+  formEl.addEventListener("submit", (e) => {
+    e.preventDefault();
+    const { value: firstName } = formEl.querySelector(
+      ".form__field--first-name"
+    );
+    const { value: lastName } = formEl.querySelector(".form__field--last-name");
+    const data = { firstName: firstName, lastName: lastName };
```

Comment:
Można zapisać krócej jak klucz i wartość to ta sama "nazwa" tj. `{firstName, lastName}`